### PR TITLE
 tools: add capability to provide make command via environment 

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
This PR enables the user to set the name of the `make` command via the `MAKE` environment variable in our test scripts. If it is not provided it defaults to just `make` in all cases.

While not strictly necessary for the use with FreeBSD, as one could bend the `gmake` command accordingly there to not use the incompatible BSD Make

https://github.com/RIOT-OS/RIOT/blob/8204931eb14c2be710b5adeb3b5ef7c7d8cdd992/dist/tools/vagrant/freebsd/Vagrantfile#L24-L28

I find it friendlier if that would be not needed to run our scripts (I can imagine similar problems might be faced on Mac OS X and Windows).

### Testing procedure
Touched tools and tests should be run 

- [x] `dist/tools/ci/print_toolchain_versions.sh`
- [x] Any test using `testrunner` (one should be enough, but we can also just run `CI: run tests`)
- [x] `dist/tools/compile_and_test_for_board/compile_and_test_for_board.py` for at least one board
- [x] `dist/tools/compile_test/compile_test.py` for at most one application combination (can be canceled after the first application passes)
- [x] `tests/lwip` on `native`
- [x] `tests/riotboot`
- [x] `tests/warn_conflicts`

### Issues/PRs references
Discovered while working on #14458, but not required for it.

### CODE REVIEW

File: dist/pythonlibs/testrunner/spawn.py [hunk 1]
```diff
@@ -41,13 +41,14 @@
 TESTRUNNER_RESET_AFTER_TERM = int(os.environ.get('TESTRUNNER_RESET_AFTER_TERM')
                                   or '0')
 
+MAKE = os.environ.get('MAKE', 'make')
```
Comment (aabadie) [inline dist/pythonlibs/testrunner/spawn.py:44]: Reported by Travis:

```suggestion
MAKE = os.environ.get('MAKE', 'make')

```

Comment (aabadie) [reply dist/pythonlibs/testrunner/spawn.py:44]: You can squash directly

**Comment by aabadie** [comment]
The [run failures on Murdock](https://ci.riot-os.org/RIOT-OS/RIOT/14459/1cb0e09d8876cf70be67aaf18b8c9a3cf5c0b877/output.html#error0) are unrelated. I think there's an issue with one of the nrf52dk workers. Re-triggering Murdock without "run tests" label.

**Comment by miri64** [comment]
Then `CI: skip compile tests` would have been also a good idea, since this is only touching test executing scripts. But oh well, now it is already running ^^

**Pull request merged**